build-tools/packages/build-cli/src/commands/release/report-unreleased.ts

```diff
@@ -0,0 +1,280 @@
+}
+
+export class UnreleasedReportCommand extends BaseCommand<typeof UnreleasedReportCommand> {
+	static description = `Creates a release report for the most recent build published to an internal ADO feed. It does this by finding the most recent build in ADO produced from a provided branch, and creates a report using that version. The report always uses the "caret" report format.`;
+
+	static flags = {
+		repo: Flags.string({
+			description: "Repository name",
+			required: true,
+		}),
+		ado_pat: Flags.string({
+			description: "ADO Personal Access Token",
+			required: true,
+			env: "ADO_PAT",
+		}),
+		sourceBranch: Flags.string({
+			description: "Branch name across which the dev release manifest should be generated.",
+			required: true,
+		}),
+		output: Flags.string({
+			description: "Output manifest file path",
+			required: true,
+		}),
+		...BaseCommand.flags,
+	};
+
+	public async run(): Promise<void> {
+		const flags = this.flags;
+
+		const repoName: string[] = flags.repo.split("/");
+
+		const PACKAGE_NAME = "@fluidframework/container-runtime";
+		const GITHUB_RELEASE_URL = `https://api.github.com/repos/${flags.repo}/releases`;
+		const ADO_BASE_URL = `https://dev.azure.com/${repoName[1].toLowerCase()}/internal/_apis/build/builds?api-version=7.0`;
+		const REGISTRY_URL = `https://pkgs.dev.azure.com/${repoName[1].toLowerCase()}/internal/_packaging/build/npm/registry/`;
+
+		// Authorization header for Azure DevOps
+		const authHeader = `Basic ${Buffer.from(`:${flags.ado_pat}`).toString("base64")}`;
+
+		// Check if the authorization header is valid
+		if (authHeader === undefined || authHeader === null) {
+			this.error("Check your ADO Personal Access Token. It maybe incorrect or expired.");
+		}
+
+		try {
+			// Get the most recent successful build number
+			const buildNumber = await getFirstSuccessfulBuild(
+				authHeader,
+				ADO_BASE_URL,
+				flags.sourceBranch,
+				this.logger,
+			);
+			if (buildNumber !== undefined) {
+				this.log(
+					`Most recent successful build number within the last 24 hours for ${flags.sourceBranch} branch: ${buildNumber}`,
+				);
+				// Fetch the dev version number
+				const devVersion = await fetchDevVersionNumber(
+					authHeader,
+					REGISTRY_URL,
+					PACKAGE_NAME,
+					buildNumber,
+					this.logger,
+				);
+				if (devVersion !== undefined) {
+					this.log(`Fetched dev version: ${devVersion}`);
+					// Generate and write the modified manifest
+					const manifestFile = await generateReleaseReportForUnreleasedVersions(
+						GITHUB_RELEASE_URL,
+						devVersion,
+						this.logger,
+					);
+					if (manifestFile !== undefined) {
+						await writeManifestToFile(manifestFile, flags.output, this.logger);
+					}
+				}
+			} else if (buildNumber === undefined) {
+				this.log(
+					`No successful build found for ${flags.sourceBranch} branch in the last 24 hours`,
+				);
+			}
+		} catch (error: unknown) {
+			this.errorLog(`Error creating manifest file: ${error}`);
+		}
```

Comment:
If you're not doing any error handling it might be better to either not catch here or rethrow.